File: Backend/src/Fibertest30.Application/SystemEvent/EventData/Monitoring/MonitoringPortScheduleChangedData.cs
using System.Text.Json;

namespace Fibertest30.Application;

public class MonitoringPortScheduleChangedData : ISystemEventData
{
    public int MonitoringPortId { get; init; }
    public MonitoringSchedulerMode Mode { get; init; }
    public int? Interval { get; init; }
    public List<int>? TimeSlotIds { get; init; }

    public MonitoringPortScheduleChangedData(int monitoringPortId, 
        MonitoringSchedulerMode mode, int? interval, List<int>? timeSlotIds)
    {
        MonitoringPortId = monitoringPortId;
        Mode = mode;
        Interval = interval;
        TimeSlotIds = timeSlotIds;
    }

    public string ToJsonData()
    {
        return JsonSerializer.Serialize(this);

    }
}

File: WebFt30/src/app/shared/system-events/system-event-data/monitoring/monitoring-port-schedule-changed-data.ts
import { MonitoringSchedulerMode } from 'src/grpc-generated';
import { Duration } from 'src/grpc-generated/google/protobuf/duration';

export interface MonitoringPortScheduleChangedData {
  MonitoringPortId: number;
  Mode: MonitoringSchedulerMode;
  Interval: number | null;
  TimeSlotIds: number[];
}
